naptime-graphql/src/test/scala/org/coursera/naptime/ari/graphql/schema/NaptimeUnionFieldTest.scala

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2016 Coursera Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.coursera.naptime.ari.graphql.schema
+
+import com.linkedin.data.schema.DataSchema
+import com.linkedin.data.schema.IntegerDataSchema
+import com.linkedin.data.schema.Name
+import com.linkedin.data.schema.RecordDataSchema
+import com.linkedin.data.schema.RecordDataSchema.Field
+import com.linkedin.data.schema.RecordDataSchema.RecordType
+import org.junit.Test
+import org.scalatest.junit.AssertionsForJUnit
+import org.scalatest.mock.MockitoSugar
+import com.linkedin.data.schema.UnionDataSchema
+import org.coursera.naptime.ari.graphql.Models
+import org.mockito.Mockito.when
+import sangria.schema.IntType
+import sangria.schema.ObjectType
+import sangria.schema.UnionType
+
+import scala.collection.JavaConverters._
+
+class NaptimeUnionFieldTest extends AssertionsForJUnit with MockitoSugar {
+
+  private[this] val resourceName = "courses.v1"
+  private[this]val schemaMetadata = mock[SchemaMetadata]
+  private[this]val resource = Models.courseResource
+  when(schemaMetadata.getResourceOpt(resourceName)).thenReturn(Some(resource))
+  when(schemaMetadata.getSchema(resource)).thenReturn(Some(null))
+
+  def buildUnionDataSchema(types: List[DataSchema], typedDefinitions: Option[Map[String, String]] = None): UnionDataSchema = {
+    val union = new UnionDataSchema()
+    val stringBuilder = new java.lang.StringBuilder()
+    union.setTypes(types.asJava, stringBuilder)
+    typedDefinitions.foreach(td => union.setProperties(
+      Map("typedDefinition" -> td.asJava.asInstanceOf[AnyRef]).asJava))
+    union
+  }
+
+  private[this] def buildRecordField(name: String, fields: List[Field]) = {
+    val fullName = new Name(name, "org.coursera.naptime", new java.lang.StringBuilder())
+    val recordDataSchema = new RecordDataSchema(fullName, RecordType.RECORD)
+    fields.foreach(_.setRecord(recordDataSchema))
+    val stringBuilder = new java.lang.StringBuilder()
+    recordDataSchema.setFields(fields.asJava, stringBuilder)
+    recordDataSchema
+  }
+
+  @Test
+  def build_SingleElementUnion() = {
+    val values = List(new IntegerDataSchema())
+    val union = buildUnionDataSchema(values)
+    val fieldName = "intOnlyUnion"
+    val field = NaptimeUnionField.build(schemaMetadata, union, fieldName, None, resourceName)
+
+    val expectedUnionTypes = List(
+      ObjectType("courses_v1_intMember", List(
```

Comment:
names for unions are a combination of the resource name and the union member name. so this is the wrapper for an int in courses.v1.

(the reason that we namespace the typenames by resource is for related resources -- if you have a child record for instance that is shared across Courses and CourseDetails, but only has a relation on the "courses.v1" resource (possibly using some field that's only available on the Course), we don't want that relationship to show up on the CourseDetails type too.